### ISSUE TITLE
infinite scrolling with grouping eats data

### ISSUE THREAD

**Issue by cstephens-cni** [inovua/reactdatagrid#328]
- what edition are you using - enterprise
- version for: your own demo

Relevant code or config

https://reactdatagrid.io/demo

What you did:

- enable infinite scrolling
- group by department (drag to group header)
- minimize first group
- repeat for all groups

What happened:

The grid eats the other group data and you can not get it back.

Reproduction repository:

https://reactdatagrid.io/demo

Problem description:

Well, it is unusable if you want to use infinite scrolling with the grouping features. I suspect this goes even deeper.  The infinite scrolling seems to not keep a model in memory.  Actually i'm not really sure how or why it would break like this.  We have developed our apps with AG Grid and I had been very excited to try out yours to see about replacing it.  I think this is going to prevent that.

Suggested solution:

https://www.ag-grid.com/example : Increase data size to 50,000 rows and Group this off language.

**Comment by cstephens-cni** [comment]
actually I'm seeing very weird behavior across the board with this combo, changing the sort order of the group seems to bring back all groups.  However, if you go to expand one it all gets eaten again.  There maybe only actually a smaller piece of code causing the issue, it may not be systemic.

**Comment by cstephens-cni** [comment]
Just wanted to update this,  Infinite scrolling to me means paging that happens automatically as you scroll (virtualized rows), I think I misinterpreted what you meant in the demo.  I think the demo means not only is there no paging but it also uses remote data.  So I maybe wrong in my thoughts on that feature, anyways.

I am sure that what I pointed out is not a desirable experience.  However, its just a demo and perhaps in trying to show everything there are corner cases that people can hit, but aren't really what you are trying to demonstrate.  I'll  close this out, since i think it's just demo related and not product related.

I am not sure if this is an issue in the real world.  I haven't hit many issues.  But I haven't had and probably won't have enough time to manufacture one to truly test it out.
